core/CMakeLists.txt

```diff
@@ -16,6 +16,10 @@ add_subdirectory(thread)
 add_subdirectory(zip)
 add_subdirectory(lzma)
 
+if(testing)
+  ROOT_ADD_TEST_SUBDIRECTORY(test)
```

Comment:
The nice thing about moving this next to src/ is that we can now systematically include the test/ subdirs in `ROOT_LINKER_LIBRARY` in cmake/modules/RootNewMacros.cmake. And yes that macro is a misnomer since a while already, it should probably be called `ROOT_ADD_MODULE`.